backend/datasource/datasource.go

```diff
@@ -0,0 +1,85 @@
+package datasource
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+// CheckDataSourceHealthRequest contains the healthcheck request
+type CheckDataSourceHealthRequest struct {
+	pluginConfig     backend.PluginConfig
+	OrgID            int64
+	DataSourceConfig backend.DataSourceConfig
+}
+
+// CheckDataSourceHealthHandler enables users to send health check
+// requests to a data source plugin.
+type CheckDataSourceHealthHandler interface {
+	CheckDataSourceHealth(ctx context.Context, req *CheckDataSourceHealthRequest) (*backend.CheckHealthResult, error)
+}
+
+type CheckDataSourceHealthHandlerFunc func(ctx context.Context, req *CheckDataSourceHealthRequest) (*backend.CheckHealthResult, error)
+
+func (fn CheckDataSourceHealthHandlerFunc) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	return fn(ctx, &CheckDataSourceHealthRequest{
+		pluginConfig:     req.PluginConfig,
+		OrgID:            req.PluginConfig.OrgID,
+		DataSourceConfig: *(req.PluginConfig.DataSourceConfig),
+	})
+}
+
+type CallDataSourceResourceRequest struct {
+	pluginConfig     backend.PluginConfig
+	OrgID            int64
+	DataSourceConfig backend.DataSourceConfig
```

Comment:
Only OrgID and DataSourceConfig exposed here compared to regular `CallResourceRequest` which includes both PluginConfig and DataSourceConfig.